pkg/reconcilers/shared/postcreate_hook.go

```diff
@@ -45,72 +45,82 @@ func (r *BaseReconciler) ReconcileUpdatePostCreateHook(ctx context.Context, hcp
     logger := clog.FromContext(ctx)
     namespace := util.GenerateNamespaceFromControlPlaneName(hcp.Name)
     
-    if hcp.Spec.PostCreateHook == nil {
-        return nil
-    }
-
-    // Hook is only applied once after CP creation
-    _, ok := hcp.Status.PostCreateHooks[*hcp.Spec.PostCreateHook]
-    if ok {
-        return nil
-    }
-
-    logger.Info("Running ReconcileUpdatePostCreateHook", "post-create-hook", *hcp.Spec.PostCreateHook)
-
-    // Get the post create hook
-    hook := &v1alpha1.PostCreateHook{
-        ObjectMeta: metav1.ObjectMeta{
-            Name: *hcp.Spec.PostCreateHook,
-        },
-    }
-    if err := r.Client.Get(context.TODO(), client.ObjectKeyFromObject(hook), hook, &client.GetOptions{}); err != nil {
-        return fmt.Errorf("error retrieving post create hook %s: %w", *hcp.Spec.PostCreateHook, err)
-    }
-
-    // Build variables with proper precedence:
-    // 1. Default vars from PostCreateHook
-    // 2. User-provided vars from ControlPlane (override defaults)
-    // 3. Built-in system vars (highest priority)
-    vars := make(map[string]interface{})
+    // Collect all hooks to process (legacy + new)
+    var hooks []v1alpha1.PostCreateHookUse
     
-    // Add default variables from hook spec
-    for _, dv := range hook.Spec.DefaultVars {
-        vars[dv.Name] = dv.Value
+    // Add legacy hook if specified
+    if hcp.Spec.PostCreateHook != nil {
+        hooks = append(hooks, v1alpha1.PostCreateHookUse{
+            HookName: hcp.Spec.PostCreateHook,
+            Vars:     hcp.Spec.PostCreateHookVars,
+        })
     }
     
-    // Override with user-provided variables from control plane
-    for key, val := range hcp.Spec.PostCreateHookVars {
-        vars[key] = val
-    }
+    // Add new hooks
+    hooks = append(hooks, hcp.Spec.PostCreateHooks...)
     
-    // Add system variables (highest priority)
-    vars["Namespace"] = namespace
-    vars["ControlPlaneName"] = hcp.Name
-    vars["HookName"] = *hcp.Spec.PostCreateHook
-
-    // Apply the hook templates
-    if err := applyPostCreateHook(ctx, r.ClientSet, r.DynamicClient, hook, vars, hcp); err != nil {
-        if util.IsTransientError(err) {
-            return fmt.Errorf("failed to apply post-create hook: %w", err) // Retry
+    for _, hook := range hooks {
+        if hook.HookName == nil {
+            logger.Info("Skipping hook with nil name")
+            continue
+        }
+        hookName := *hook.HookName
+        
+        // Skip already applied hooks
+        if hcp.Status.PostCreateHooks != nil && hcp.Status.PostCreateHooks[hookName] {
+            continue
+        }
+        
+        logger.Info("Processing post-create hook", "hook", hookName)
+        
+        // Get hook definition
+        pch := &v1alpha1.PostCreateHook{}
+        if err := r.Client.Get(ctx, client.ObjectKey{Name: hookName}, pch); err != nil {
```

Comment:
On transient errors, the function immediately returns, preventing other hooks from being processed.  Would be better accumulating errors and reporting them collectively, or ensure that a failed hook (transient or not) doesn’t block other hooks from being attempted